static/app/views/detectors/detectorListTable.tsx

```diff
@@ -0,0 +1,193 @@
+import {useCallback, useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+import {type AnimationProps, motion} from 'framer-motion';
+
+import {Button} from 'sentry/components/button';
+import Checkbox from 'sentry/components/checkbox';
+import {Flex} from 'sentry/components/container/flex';
+import Panel from 'sentry/components/panels/panel';
+import PanelBody from 'sentry/components/panels/panelBody';
+import PanelHeader from 'sentry/components/panels/panelHeader';
+import {IconArrow} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {type Detector, DetectorListRow} from 'sentry/views/detectors/detectorListRow';
+
+const animationProps: AnimationProps = {
+  initial: {translateY: 8, opacity: 0},
+  animate: {translateY: 0, opacity: 1},
+  exit: {translateY: -8, opacity: 0},
+  transition: {duration: 0.1},
+};
+
+type DetectorListTableProps = {
+  detectors: Detector[];
+};
+
+function DetectorListTable({detectors}: DetectorListTableProps) {
+  const [direction, setDirection] = useState<'asc' | 'desc'>('desc');
+  const [selectedRows, setSelectedRows] = useState<string[]>([]);
+
+  const detectorIds = useMemo(() => detectors.map(detector => detector.id), [detectors]);
+
+  const handleSelect = useCallback(
+    (id: string, checked: boolean): void => {
+      if (checked) {
+        setSelectedRows([...selectedRows, id]);
+      } else {
+        setSelectedRows(selectedRows.filter(item => item !== id));
+      }
+    },
+    [selectedRows]
+  );
+
+  function selectAll(): void {
```

Comment:
Since this toggles the `selectAll` state, I'd recommend using `toggleSelectAll()`